src/example/ERC20LockerExample.sol

```diff
@@ -80,17 +97,33 @@ contract ERC20LockerExample is ERC20Locker {
         emit Withdrawn(msg.sender, amount);
     }
 
+    function registerApproval(uint256 service, uint256 amount) internal {
+        require(!_stakingHub.isFrozen(msg.sender), "Staker is frozen");
+        bool decreasing = _registerApproval(msg.sender, service, amount);
+        if (decreasing) {
+            _totalStakes[service] -= _calcStakeDecreaseForAllowanceChange(_balances[msg.sender], _allowances[msg.sender][service].allowance, amount);
+        } else {
```

Comment:
This is not quite like depositing because there's no guarantee the staker will approve the new (increased) allowance.

Maybe we should do the both cases on finalization?

Reply:
Changed in https://github.com/0xPolygon/staking-hub/pull/3/commits/a5184ed046ffe6ac51ad593e41d1e1ce7db0ac97.